apps/dash-manufacture-spc-dashboard/app.py

```diff
@@ -0,0 +1,1209 @@
+                "x0": 1,
+                "y0": mean,
+                "x1": len_figure + 1,
+                "y1": mean,
+                "line": {"color": "rgb(255,127,80)", "width": 2},
+            },
+            {
+                "type": "line",
+                "xref": "x",
+                "yref": "y",
+                "x0": 1,
+                "y0": lcl,
+                "x1": len_figure + 1,
+                "y1": lcl,
+                "line": {"color": "rgb(255,127,80)", "width": 1, "dash": "dashdot"},
+            },
+        ],
+        xaxis2={
+            "title": "Count",
+            "domain": [0.8, 1],  # 70 to 100 % of width
+            "titlefont": {"color": "darkgray"},
+            "showgrid": False,
+        },
+        yaxis2={
+            "anchor": "free",
+            "overlaying": "y",
+            "side": "right",
+            "showticklabels": False,
+            "titlefont": {"color": "darkgray"},
+        },
+    )
+
+    return fig
+
+
+def update_sparkline(interval, param):
+    x_array = state_dict["Batch"]["data"].tolist()
+    y_array = state_dict[param]["data"].tolist()
+
+    if interval == 0:
+        x_new = y_new = None
+
+    else:
+        if interval >= max_length:
+            total_count = max_length
+        else:
+            total_count = interval
+        x_new = x_array[:total_count][-1]
+        y_new = y_array[:total_count][-1]
+
+    return dict(x=[[x_new]], y=[[y_new]]), [0]
+
+
+def update_count(interval, col, data):
+    if interval == 0:
+        return "0", "0.00%", 0.00001, theme["primary"]
+
+    if interval > 0:
+
+        if interval >= max_length:
+            total_count = max_length - 1
+        else:
+            total_count = interval - 1
+
+        ooc_percentage_f = data[col]["ooc"][total_count] * 100
+        ooc_percentage_str = "%.2f" % ooc_percentage_f + "%"
+
+        # Set maximum ooc to 15 for better grad bar display
+        if ooc_percentage_f > 15:
+            ooc_percentage_f = 15
+
+        if ooc_percentage_f == 0.0:
+            ooc_grad_val = 0.00001
+        else:
+            ooc_grad_val = float(ooc_percentage_f)
+
+        # Set indicator theme according to threshold 5%
+        if 0 <= ooc_grad_val <= 5:
+            color = theme["primary"]
+        elif 5 < ooc_grad_val < 7:
+            color = "#f4d44d"
+        else:
+            color = "#FF0000"
+
+    return str(total_count + 1), ooc_percentage_str, ooc_grad_val, color
+
+
+app.layout = html.Div(
+    id="big-app-container",
+    children=[
+        build_banner(),
+        dcc.Interval(
+            id="interval-component",
+            interval=2 * 1000,  # in milliseconds
+            n_intervals=50,  # start at batch 50
+            disabled=True,
+        ),
+        html.Div(
+            id="app-container",
+            children=[
+                build_tabs(),
+                # Main app
+                html.Div(id="app-content", className="container scalable"),
```

Comment:
Wondering effect `className="container scalable"` has? I could be wrong, I just don't visually see the impact and I noticed that the properties mentioned under `.container.scalable` as shown in the css:
```
.container.scalable {
  width: 95%;
  max-width: none;
}
```
are also mentioned under `#app-content` as shown in the css:
```
#app-content {
  background: inherit;
  padding: 0;
  width: 100%;
  max-width: 100%;
}
```

Reply:
I've also seen a complete overwrite and we don't need max-width scalable for app-content, will remove.